### PR TITLE
Fix getPid failure in Windows

### PR DESCRIPTION
**Pull request by Ailloviee**
Revert changes made previously on #1638

Fix getPid test script path failture [#1630](https://github.com/eclipse/openj9/issues/1630)

Added test case in getPidTest.pl for getting pid in Windows.

For GetPid.java: 
Added `System.in.read()` in the code to wait for input from test script(`getPidTest.pl`) where the script will write to java process(`GetPid.java`) after it gets process id from the java process. (so that the process does not terminate too fast before perl script can catch the pid).

Signed-off-by: Charles_Zheng <Juntian.Zheng@ibm.com>

### CODE REVIEW

File: test/functional/cmdLineTests/runtimemxbeanTests/src/GetPid.java [hunk 1]
```diff
@@ -23,12 +23,20 @@
...
-	}
+        private static Object monitor = new Object();
+ 
+        public static void main(String args[]) {
+                synchronized (monitor) {
+                        try {
+                                RuntimeMXBean mb = ManagementFactory.getRuntimeMXBean();
+                                long id = mb.getPid();
+                                System.out.println(id);
+                                monitor.wait(5000);
```
Comment (tajila) [inline test/functional/cmdLineTests/runtimemxbeanTests/src/GetPid.java]: I think using a `Thread.sleep(...)` is more clear than the more complicated monitor sync/wait and achieves the same goal.  
  
Also, add a comment saying why the delay is required

File: test/functional/cmdLineTests/runtimemxbeanTests/src/GetPid.java [hunk 2]
```diff
@@ -23,12 +23,20 @@
 import java.lang.management.RuntimeMXBean;
 import java.lang.management.ManagementFactory;
 
-class GetPid {
+class GetPid extends Thread {
```
Comment (tajila) [inline test/functional/cmdLineTests/runtimemxbeanTests/src/GetPid.java]: don't need to extend Thread

File: test/functional/cmdLineTests/runtimemxbeanTests/getPidTest.pl [hunk 3]
```diff
@@ -30,9 +30,28 @@
...
+	open3($in, $out, $err, $javaCmd);
+	$javaPid = <$out>;
+	$perlPid = `wmic process where "name='java.exe'" get ProcessID`;
+	print $in "getPid finished";
+	$javaPid =~ s/^\s+|\s+$//g;
+	$perlPid =~ s/^\s+|\s+$//g;
+	if (index($perlPid, $javaPid) != -1) {
+	$perlPid = $javaPid;
+	}
+	else{
```
Comment (tajila) [inline test/functional/cmdLineTests/runtimemxbeanTests/getPidTest.pl]: can remove the else

File: test/functional/cmdLineTests/runtimemxbeanTests/getPidTest.pl [hunk 4]
```diff
@@ -30,9 +30,28 @@
...
+
+if ($^O eq 'cygwin') {
+	open3($in, $out, $err, $javaCmd);
+	$javaPid = <$out>;
+	$perlPid = `wmic process where "name='java.exe'" get ProcessID`;
+	print $in "getPid finished";
+	$javaPid =~ s/^\s+|\s+$//g;
+	$perlPid =~ s/^\s+|\s+$//g;
+	if (index($perlPid, $javaPid) != -1) {
+	$perlPid = $javaPid;
```
Comment (tajila) [inline test/functional/cmdLineTests/runtimemxbeanTests/getPidTest.pl]: need an indent inside the if block

File: test/functional/cmdLineTests/runtimemxbeanTests/getPidTest.pl [hunk 5]
```diff
@@ -30,9 +30,28 @@
...
 
-my $javaPid = <$out>;
+my $javaPid;
+
+if ($^O eq 'cygwin') {
+	open3($in, $out, $err, $javaCmd);
+	$javaPid = <$out>;
+	$perlPid = `wmic process where "name='java.exe'" get ProcessID`;
+	print $in "getPid finished";
+	$javaPid =~ s/^\s+|\s+$//g;
```
Comment (tajila) [inline test/functional/cmdLineTests/runtimemxbeanTests/getPidTest.pl]: add a comment saying what this line does

File: test/functional/cmdLineTests/runtimemxbeanTests/src/GetPid.java [hunk 6]
```diff
@@ -24,11 +24,14 @@
...
-	}
-
+        public static void main(String args[]) {
+                        try {
+                                RuntimeMXBean mb = ManagementFactory.getRuntimeMXBean();
+                                long id = mb.getPid();
+                                System.out.println(id);
+                                /* Puts Java process on hold until system receives output from testing script getPidTest.pl*/
+                                System.in.read();
+                        } catch (Exception e) {
```
Comment (tajila) [inline test/functional/cmdLineTests/runtimemxbeanTests/src/GetPid.java]: some issues with indenting here, you can use the eclipse auto-format to fix this

**Comment by tajila** [comment]
> For GetPid.java:
Added Thread.sleep() in code to put the java process on wait so the test script would have time to capture process id before it terminates.

Also modify the message above for the new changes

File: test/functional/cmdLineTests/runtimemxbeanTests/getPidTest.pl [hunk 7]
```diff
@@ -30,9 +30,27 @@
...
+	$perlPid = `wmic process where "name='java.exe'" get ProcessID`;
+	print $in "getPid finished";
+	# String trim both sides of javaPid and perlPid for the index check
+	$javaPid =~ s/^\s+|\s+$//g;
+	$perlPid =~ s/^\s+|\s+$//g;
+	if (index($perlPid, $javaPid) != -1) {
+		$perlPid = $javaPid;
+	}
+}
+else {
```
Comment (tajila) [inline test/functional/cmdLineTests/runtimemxbeanTests/getPidTest.pl]: minor thing, put this else on the same line as `}`  
  
`} else {`

File: test/functional/cmdLineTests/runtimemxbeanTests/src/GetPid.java [hunk 8]
```diff
@@ -24,11 +25,17 @@
...
-		RuntimeMXBean mb = ManagementFactory.getRuntimeMXBean();
-		long id = mb.getPid();
-		System.out.println(id);
+	public static void main(String args[]) {
+		try {
+			RuntimeMXBean mb = ManagementFactory.getRuntimeMXBean();
+			long id = mb.getPid();
+			System.out.println(id);
+			/*
+			 * Puts Java process on hold until system receives output from
```
Comment (tajila) [inline test/functional/cmdLineTests/runtimemxbeanTests/src/GetPid.java]: receives *input* from

**Comment by tajila** [comment]
@pshipton changes are ready for another review

File: test/functional/cmdLineTests/runtimemxbeanTests/getPidTest.pl [hunk 9]
```diff
@@ -30,9 +30,26 @@
...
+
+if ($^O eq 'cygwin') {
+	open3($in, $out, $err, $javaCmd);
+	$javaPid = <$out>;
+	$perlPid = `wmic process where "name='java.exe'" get ProcessID`;
+	print $in "getPid finished";
+	# String trim both sides of javaPid and perlPid for the index check
+	$javaPid =~ s/^\s+|\s+$//g;
+	$perlPid =~ s/^\s+|\s+$//g;
+	if (index($perlPid, $javaPid) != -1) {
```
Comment (pshipton) [inline test/functional/cmdLineTests/runtimemxbeanTests/getPidTest.pl]: Please comment to explain this code.

File: test/functional/cmdLineTests/runtimemxbeanTests/src/GetPid.java [hunk 10]
```diff
@@ -24,11 +25,17 @@
...
+		try {
+			RuntimeMXBean mb = ManagementFactory.getRuntimeMXBean();
+			long id = mb.getPid();
+			System.out.println(id);
+			/*
+			 * Puts Java process on hold until system receives input from
+			 * testing script getPidTest.pl
+			 */
+			System.in.read();
+		} catch (Exception e) {
```
Comment (pshipton) [inline test/functional/cmdLineTests/runtimemxbeanTests/src/GetPid.java]: Better to put the try catch just around the System.in.read() so exceptions are ignored only for this call.

**Comment by llxia** [comment]
- LGTM. Removing `D=\\` will affect all tests that use ${D} on wins. A complete testing on wins is needed.
- Please squash all commits into one and put `[ci skip]` in the commit msg to skip sdk build (travis ci).

**Comment by DanHeidinga** [comment]
@llxia what needs to happen for this to be merged?

**Comment by Ailloviee** [comment]
@llxia I just launched a build with correct omr and openj9 version yesterday. Previously tests were failing due to inconsistency between the two. The tests are still going now, should be done by noon.

**Comment by llxia** [comment]
Jenkins test sanity,extended win jdk10

**Comment by DanHeidinga** [comment]
@llxia Is this ready to merge?

**Pull request merged**